### PR TITLE
Call api from frontend to for CRUD in side nav bar

### PR DESCRIPTION
**Pull request by luzivaltic**

### CODE REVIEW

File: app/components/SideNavBar.tsx [hunk 1]
```diff
@@ -1,17 +1,115 @@
...
+  title: string;
+};
+
+type Chapter = {
+  id: string;
+  title: string;
+  subject_id: string;
+};
+
+interface CommonHeaderProperties extends HeadersDefaults {
```
Comment (nolannguyen1212) [inline app/components/SideNavBar.tsx:27]: why using interface instead of type ?

File: app/components/SideNavBar.tsx [hunk 2]
```diff
@@ -1,17 +1,115 @@
...
-  const listSubject = [
-    "Network safety and Security",
-    "Computer Graphics",
-    "Network Programming",
-    "Web Application Development",
-  ];
+  const cookies = useCookies();
+  const [listSubject, setListSubject] = useState<Subject[]>([]);
+  const [listChapter, setListChapter] = useState<Chapter[]>([]);
+  const [chosenSubject, setChosenSubject] = useState(10000);
```
Comment (nolannguyen1212) [inline app/components/SideNavBar.tsx:35]: why initial subject is set equal 10000 ?

File: app/components/SideNavBar.tsx [hunk 3]
```diff
@@ -1,17 +1,115 @@
...
+
+  axios.defaults.headers.common["Authorization"] = `Bearer ${cookies.get(
+    "access_token"
+  )}`;
+
+  const handleClickSubject = (index: number) => {
+    if (chosenSubject == index) {
+      setChosenSubject(10000);
```
Comment (nolannguyen1212) [inline app/components/SideNavBar.tsx:43]: what is 10000 ?

File: app/components/SideNavBar.tsx
Comment (natalieconan) [inline app/components/SideNavBar.tsx:1]: Can we move all function like Create / Get List Chapter to another files like ApiStore ? It'll be reused later.

**Review by william1nguyen — CHANGES_REQUESTED** [review]

**Review by william1nguyen — APPROVED** [review]

**Pull request merged**